web/src/components/data-entities/diver/DiverList.js

```diff
@@ -1,103 +1,129 @@
-import React, {useEffect, useState} from 'react';
+import React, {useState, useRef} from 'react';
 import {Box, Button, Typography} from '@mui/material';
 import {NavLink} from 'react-router-dom';
 import {grey, red} from '@mui/material/colors';
 import {getResult, entityEdit} from '../../../api/api';
 import LoadingOverlay from '../../overlays/LoadingOverlay';
 import {AgGridColumn, AgGridReact} from 'ag-grid-react';
 import {Add, Save} from '@mui/icons-material';
+import {constants} from '../../../common/constants';
 import 'ag-grid-enterprise';
 
 const DiverList = () => {
-  const [gridApi, setGridApi] = useState();
-  const [delta, setDelta] = useState([]);
-  const [rowData, setRowData] = useState();
-  const [errors, setErrors] = useState([]);
+    const [delta, setDelta] = useState([]);
+    const [errors, setErrors] = useState([]);
+    const dGridRef = useRef(null);
 
-  useEffect(() => {
-    async function fetchDivers() {
-      await getResult('divers').then((res) => setRowData(res.data));
-    }
-    fetchDivers();
-  }, []);
+    // Auto size function to be call each time data changed, so the grid always autofit
+    const autoSizeAll = (skipHeader) => {
+        if (dGridRef.current != null) {
```

Comment:
should be `typeof dGridRef !== "null"`, but you can also use null coalescing to write at dGridRef?.current, or just use a truthy check `if(dGridRef.current) {`

Reply:
Updated